### ISSUE TITLE
Serde for all curves

### ISSUE THREAD

**Issue by umgefahren** [RustCrypto/elliptic-curves#393]
In order to utilize this library in my library it would be amazing if at least PublicKey would implement Serialize and Deserialize. The underlaying elliptical-curve crate seems to have a serde feature. However I don't not if this is enough to implement these traits for all curves. Currently I'm working with a very ugly workaround.

**Comment by tarcieri** [comment]
The `PublicKey` type is shared by all curve implementations (with arithmetic anyway, which is presently `k256` and `p256), so adding support in the `elliptic-curve` crate will benefit them all.

`EncodedPoint` is another type which would benefit all curves, and not just the ones with arithmetic implementations.

**Comment by newpavlov** [comment]
IIRC we had a discussion about it in Zulip. I think that we should be really careful with implementing `serde` traits for `PublicKey`. Note that a naive derive will not work, since inner representation of `PublicKey` depends on target architecture. So a hypothetical implementation would have to convert from/to bytes first and only then serialize/deserialize it.

**Comment by tarcieri** [comment]
Yeah, it should convert to an `EncodedPoint` first as part of decoding/encoding. Then there's the problem that different serde format implementations have different opinions on how to handle bytes, so at that point you're probably looping in a crate like `serde_bytes`.

**Comment by tarcieri** [comment]
We should now have ubiquitous `serde` support. Please comment and/or reopen this issue if you see any gaps.
